src/chess/board.cc

```diff
@@ -719,6 +733,41 @@ bool ChessBoard::IsUnderAttack(BoardSquare square) const {
   return false;
 }
 
+bool ChessBoard::IsSameMove(Move move1, Move move2) const {
```

Comment:
The way this is written, "e1a1", "e1b1" and "e1c1" are considered the same.

Reply:
Only if they all are king's castlings, right? And it's not possible for all of them to be legal king's castlings.

Reply:
I meant for normal chess, so only "e1c1" and "e1a1" are legal but "e1b1" is also accepted. What happens is that moving the king two or more places is considered a castling move, even if there is no rook there. This way the "e1c1" notation works OK, but maybe this needs to be stricter, and only return true for `IsSameMove("e1c1", "e1a1")` and `IsSameMove("e1g1", "e1h1")` (and vice versa).